custom_components/waste_collection_schedule/waste_collection_schedule/source/landkreis_helmstedt_de.py

```diff
@@ -0,0 +1,170 @@
+                break
+            
+        if not foundMunicipalCalendarICSDownloadURL:
+            raise SourceArgumentException(argument="municipal", message="No Abholgebiet found for " + self._municipal + ". Please check " + API_URL + " for correct ICS calendar name.")
+
+        r = requests.get(foundMunicipalCalendarICSDownloadURL, headers=HEADERS)
+        r.raise_for_status()
+
+        dates = self._ics.convert(r.text)
+        elements = []
+
+        missingAltpapier = True
+        missingBioabfall = True
+        missingGelberSack = True
+        missingRestabfall = True
+
+        for d in dates:
+            type = d[1]
+            date = d[0]
+            pickUpArea = type[-1:]
+            pickUpType = type[:-2]
+
+            addDate = False
+            if pickUpType == COLLECTION_TYPE_ALTPAPIER:
+                if pickUpArea == str(self._altpapier):
+                    addDate = True
+                    missingAltpapier = False
+            elif pickUpType == COLLECTION_TYPE_BIOABFALL:
+                if pickUpArea == str(self._bioabfall):
+                    addDate = True
+                    missingBioabfall = False
+            elif pickUpType == COLLECTION_TYPE_GELBER_SACK:
+                if pickUpArea == str(self._gelber_sack):
+                    addDate = True
+                    missingGelberSack = False
+            elif pickUpType == COLLECTION_TYPE_RESTABFALL:
+                if pickUpArea == str(self._restabfall):
+                    addDate = True
+                    missingRestabfall = False
+    
+            if addDate:
+                elements.append(
+                    Collection(
+                        date = date,  # Collection date
+                        t = pickUpType,  # Collection type
+                        icon = ICON_MAP.get(pickUpType),  # Collection icon
+                    )
+                )
+
+        self.abort_when_missing_collection_types(missingAltpapier, missingBioabfall, missingGelberSack, missingRestabfall)
+        
+        return elements
+
+    def abort_when_missing_collection_types(self, missingAltpapier, missingBioabfall, missingGelberSack, missingRestabfall):
+        if(missingGelberSack):
+            raise SourceArgumentNotFoundWithSuggestions(argument="gelber_sack", value=self._gelber_sack, suggestions=["Check PDF calendar for valid collection areas."])
+        if(missingAltpapier):
+            raise SourceArgumentNotFoundWithSuggestions(argument="altpapier", value=self._altpapier, suggestions=["Check PDF calendar for valid collection areas."])
+        if(missingBioabfall):
+            raise SourceArgumentNotFoundWithSuggestions(argument="bioabfall", value=self._bioabfall, suggestions=["Check PDF calendar for valid collection areas."])
+        if(missingRestabfall):
+            raise SourceArgumentNotFoundWithSuggestions(argument="restabfall", value=self._restabfall, suggestions=["Check PDF calendar for valid collection areas."])
```

Comment:
`SourceArgumentNotFoundWithSuggestions` does not what you think it does. It provides additional suggestions to the user. So in this scenario the error occurs with the message "we could not find values for the argument [...]" and adds a drop-down menu showing "check PDF calendar [...]" the suggestion should be valid alternative values, not an error message.

The error for use case is `SourceArgumentException`